x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -93,18 +95,22 @@ export const createLogThresholdExecutor = (libs: InfraBackendLibs) =>
     LogThresholdAlertState,
     LogThresholdAlertContext,
     LogThresholdActionGroups
-  >(async ({ services, params }) => {
-    const { alertWithLifecycle, savedObjectsClient, scopedClusterClient } = services;
-    const { sources } = libs;
-    const alertFactory: LogThresholdAlertFactory = (id, reason, value, threshold) =>
-      alertWithLifecycle({
+  >(async ({ services, params, startedAt }) => {
+    const { alertWithLifecycle, savedObjectsClient, scopedClusterClient, getAlertStartedDate } =
+      services;
+    const { sources, kibanaBaseUrl } = libs;
+    let indexedStartedAt: string | null = startedAt.toISOString();
+    const alertFactory: LogThresholdAlertFactory = (id, reason, value, threshold) => {
+      indexedStartedAt = getAlertStartedDate(id) ?? indexedStartedAt;
```

Comment:
@fkanout I am just trying to understand where every variable is coming from. Correct me if I am wrong:
- `startedAt` comes from the alerting framework (the task executor/scheduler whatever it is called)
- `getAlertStartedDate` comes from the rule_registry

Reply:
Correct. The rule registry takes the `startedAt` in the first executor run and used it with ALERT_STARTED and indexes it. `getAlertStartedDate` returns that indexed `startedAt` value.

Reply:
Awesome thanks a lot for great explanation!

Reply:
No problem!